### ISSUE TITLE
Explore Probes & Drugs portal as a source for Chemical Probes

### ISSUE THREAD

**Issue by ireneisdoomed** [opentargets/platform#1536]
1. We want to explore the PostgreSQL dump from probes-drugs.com to build a dataset that would include the sources listing chemical probes. This could replace our current approach and will be a separate dataset from the main target object.

2. We want also want to make sure that this data is freely accessible and of public use.

Reference preprint: https://chemrxiv.org/articles/preprint/Will_the_Chemical_Probes_Please_Stand_Up_/14511666?file=27802503

**Comment by ireneisdoomed** [comment]
Probes and Drugs integrates 12 resources from which chemical probes are systematically and periodically fetched and analysed.

I've explored the Probes and Drugs source by downloading their SQL dump and navigating the tables to see how the data is connected.

So far I've been able to reproduce almost the entirety of the schema we have currently in production and have added some more fields that are available via this resource. All the changes are tracked in the schema (iter 3):
https://docs.google.com/spreadsheets/d/1AqC6aqKgyf_s-R1LculocodpjcHRMw6t-pUoVsvryxs/edit#gid=719153991

Some points for further discussion:
- [ ] In the referenced preprint where an analysis of the different probes sources is made, `MLP` and `Nature Chemical Biology Probes` are considered to include not high-quality probes (legacy). They are legacy collections that have a negative weight in their proposed scoring. Should we include them?
- [ ] Which resource we should be linking out? Probes and Drugs or the original sources?
- [ ] Target IDs: 
   - 1. Is it better to map using Uniprot IDs or Gene Symbols?
   - 2. Do we want to keep target name? Might be useful if the probe targets a protein family.  
- [ ] Extra field to distinguish between experimental and calculated probes
- [ ] How granular do we want to be with the scoring system?

Pending actions to be done:
- [ ] Map probe and target to our indexes.
- [ ] Extract the `in vivo` label.
- [ ] Currently we apply a filter in ProbeMiner for those targets where target_potency_raw > 5. I need to see how Probes and Drugs handles this.

**Comment by ireneisdoomed** [comment]
@andrewhercules has pointed that whenever `urls.niceName == "High-quality chemical probes"`, the field `urls.url`is missing.

This is because this is not strictly a data source but a curated subset indicating the probe is of high quality.

The proposed solution is:
1. to include a `isHighQuality` flag to capture those probes which are in the high quality subset
2. to remove `High-quality chemical probes` as a possible source for chemical probes

I have also noticed a bug in the current dataset where sources under the `urls` array are duplicated. I'll be fixing that in the next iter as well.

**Comment by d0choa** [comment]
It does look to me that a `isHighQuality` is a duplication of already available information `urls.niceName == "High-quality chemical probes"`. I might not have all the context but make sure you are not artificially duplicating information

**Comment by ireneisdoomed** [comment]
Sorry I didn't phrase it well enough.

probes-drugs.org organises the probes into **subsets**, which always coincide with the provenance of the data (Chemical Probes.org,...). With the one exception of `High-quality chemical probes`, which is a curated subset drawn from all the data sources, i.e. this subset does not bring anything new. 

 `isHighQuality` won't be a duplication because `High-quality chemical probes` will no longer be an option for `urls.niceName`. I will just report the same information in a different way.

**Comment by andrewhercules** [comment]
@ireneisdoomed I also noticed some data duplication with the [API query for PIK3CA (ENSG00000121879)](https://api-beta-dot-open-targets-eu-dev.ew.r.appspot.com/api/v4/graphql/browser?query=query%20targetInfo%20%7B%0A%20%20target%28ensemblId%3A%20%22ENSG00000121879%22%29%20%7B%0A%20%20%20%20id%0A%20%20%20%20approvedSymbol%0A%20%20%20%20chemicalProbes%20%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%20%20control%0A%20%20%20%20%20%20drugId%0A%20%20%20%20%20%20inchiKey%0A%20%20%20%20%20%20mechanismOfAction%0A%20%20%20%20%20%20origin%0A%20%20%20%20%20%20probeMinerScore%0A%20%20%20%20%20%20probesDrugScore%0A%20%20%20%20%20%20scoreInCells%0A%20%20%20%20%20%20scoreInOrganisms%0A%20%20%20%20%20%20targetFromSourceId%0A%20%20%20%20%20%20urls%20%7B%0A%20%20%20%20%20%20%20%20niceName%0A%20%20%20%20%20%20%20%20url%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A).

After running the query, I see 4 duplicate entries for the drug ALPELISIB with the only difference being the `mechanismOfAction` field and the `origin`:

1 entry where `mechanismOfAction` is `null` and `origin` is `experimental`
1 entry where `mechanismOfAction` is `null` and `origin` is `calculated`
1 entry where `mechanismOfAction` is `inhibitor` and `origin` is `experimental`
1 entry where `mechanismOfAction` is `inhibitor` and `origin` is `calculated`

![Screenshot 2021-08-11 at 12 49 47](https://user-images.githubusercontent.com/7490258/129024122-d176aa71-0c12-4f94-a262-ef20f9b92ae7.png)

**Comment by ireneisdoomed** [comment]
I have addressed all the above comments in a new iteration of chemicalProbes (hopefully the last one!):
- `High-quality chemical probes` have been removed as a possible data source
- `isHighQuality` boolean flag has been added to the schema.
- The uniqueness of the fields have been revised in order to avoid explosion such as @andrewhercules raised. Now the unique fields are the fields `id`, `targetFromSourceId`, `drugId`, `inchiKey`, `control`. 
- Some fields that used to be a string are now an array: `origin`, `mechanismOfAction`.
- The score fields are the maximum value of the collected array.
- Some cleaning steps on the `control` column were applied.

The new file can be found here: `gs://otar001-core/ProbeMiner/annotation/chemicalprobes-2021-08-12.json`

There have been some changes to the schema that need to be updated in the ETL (@JarrodBaker). Those are tagged in orange: https://docs.google.com/spreadsheets/d/1AqC6aqKgyf_s-R1LculocodpjcHRMw6t-pUoVsvryxs/edit#gid=719153991

**Comment by ireneisdoomed** [comment]
I have uploaded an updated version of the dataset (`gs://otar001-core/ProbeMiner/annotation/chemicalprobes-2021-08-23.json`) with minor changes:
- [X] Replaced `Probe Miner (suitable probes)` source name to `Probe Miner`
- [X] Built more complete URLs whenever this can be handled with the data. We have comprehensive links for:`Probe Miner`, `Chemical Probes.org`, `opnMe Portal`, `Open Science Probes`, and `SGC Probes`.

I have removed the reference for `Gray Laboratory Probes` since their main website is down.

**Comment by ireneisdoomed** [comment]
I have uploaded an updated version of the dataset(`gs://otar001-core/ProbeMiner/annotation/chemicalprobes-2021-09-17.json`) to address a bugfix where there were **probes without a reference**.

The offending cases were only being referenced by the subset "High-Quality Chemical Probe", which as explained above, rather than a data source it should be a curated subset drawn from all the other data sources.
In the process of converting this source into a tag and removing the reference, I have noticed that there are some compounds where the `urls` array is null.

I have contacted P&Ds to clarify why these do not belong to a bigger set. In the meantime, such cases are referenced to Probes and Drugs as follows:
```
"urls":[{"niceName":"Probes & Drugs Portal","url":"https://www.probes-drugs.org"}]
```
